### DISCUSSION TITLE
Fatal error: Declaration of Mpdf\PsrHttpMessageShim\Request::withRequestTarget(string $requestTarget) must be compatible with PsrExt\Http\Message\RequestInterface::withRequestTarget($requestTarget) in /vendor/mpdf/psr-http-message-shim/src/Request.php on line 88

### DISCUSSION THREAD

**Discussion by programadorwebfreelance** [mpdf/mpdf#2063] [General]
I have problem generating pdf.
Fatal error: Declaration of Mpdf\PsrHttpMessageShim\Request::withRequestTarget(string $requestTarget) must be compatible with PsrExt\Http\Message\RequestInterface::withRequestTarget($requestTarget) in /vendor/mpdf/psr-http-message-shim/src/Request.php on line 88

My php version is 8.3
I use phalcon 5.8.

composer mpdf:

mpdf/mpdf                  8.2.4 PHP library generating PDF files from UTF-8 encoded HTML
mpdf/psr-http-message-shim 2.0.1 Shim to allow support of different psr/message versions.
mpdf/psr-log-aware-trait   3.0.0 Trait to allow support of different psr/log versions.

composer psr:

psr/container        2.0.2 Common Container Interface (PHP FIG PSR-11)
psr/event-dispatcher 1.0.0 Standard interfaces for event handling.
psr/http-client      1.0.3 Common interface for HTTP clients
psr/http-factory     1.1.0 PSR-17: Common interfaces for PSR-7 HTTP message factories
psr/http-message     2.0   Common interface for HTTP messages
psr/log              3.0.2 Common interface for logging libraries
psr/simple-cache     3.0.0 Common interfaces for simple caching

Thanks

**Comment by finwe** [comment]
Where does the `PsrExt` namespace come from? Does removing the vendor folder and running composer update/install help?

**Comment by melendez-dev** [comment]
Delete vendor folder and composer.lock file, run composer install again. it should fix the problem

**Reply by michalcharvat** [reply]
That didnt solve the issue. I am using the latest version of mpdf + shim on PHP 8.3.24 

The problem appears if you have installed php-psr extension.

**Reply by finwe** [reply]
Good catch. OK, guess for now, mPDF is not compatible with `php-psr` extension.
